### DISCUSSION TITLE
Do you have plans to support Java Records.

### DISCUSSION THREAD

**Discussion by moxit0** [ozlerhakan/poiji#298] [General]
Hi, good job you have done. Do you have plans in future to support mapping excel files to new java record classes?

**Comment by ozlerhakan** [comment]
Hi @moxit0 , we should consider a MAJOR release for Poiji to be compatible with Java 14 and then the record feature can be on our radar.

**Comment by moxit0** [comment]
Hi, will definitely try it soon and will give you feedback. Thank you.

**Comment by moxit0** [comment]
Hi, tested with java records. Works perfect for our simple cases. Great job
guys.

Thank you.

На нд, 17.03.2024 г. в 14:38 vaa25 ***@***.***> написа:

**Comment by steveschols** [comment]
@moxit0 Can you give an example how?
I'm getting the following when using Java records as DTO to map an Excel file to.
`Caused by: java.lang.NoSuchMethodException: com.poc.xlsximport.excel.ExcelImport.<init>()
	at java.base/java.lang.Class.getConstructor0(Class.java:3585)
	at java.base/java.lang.Class.getDeclaredConstructor(Class.java:2754)
	at com.poiji.util.ReflectUtil.newInstanceOf(ReflectUtil.java:23)`

**Comment by ozlerhakan** [comment]
Java Records are part of the major version 5.0.0 of poiji. Thanks!
